lib/engine/game/base.rb

```diff
@@ -2559,7 +2559,7 @@ def ability_usable_this_or?(ability)
       def ability_right_time?(ability, time, on_phase, passive_ok, strict_time)
         return true unless @round
         return true if time == 'any' || ability.when?('any')
-        return (on_phase == ability.on_phase) || (on_phase == 'any') if on_phase
+        return (on_phase == ability.on_phase) || (on_phase == 'any') if ability.on_phase
```

Comment:
Shouldn't it be ability.on_phase == 'any' as well then?

Reply:
I did test it, but it provoked some unexpected problems: the following ability didn't work.
```
              {
                type: 'close',
                on_phase: 'never',
                owner_type: 'corporation',
              },
```
Possibly removing the whole second part, i.e. ` || (on_phase == 'any')` could make sense though. I'll check later how it behaves.

Reply:
will you check this?

Reply:
I plan to do it tomorrow, not having time today.

Reply:
I tested it. Both

 `return (on_phase == ability.on_phase) if ability.on_phase`

and

`return (on_phase == ability.on_phase) || (ability.on_phase == 'any') if ability.on_phase`

result in the following ability not working:
```             
              {
                type: 'close',
                on_phase: 'never',
                owner_type: 'corporation',
              },
```

Whilst with

`return (on_phase == ability.on_phase) || (on_phase == 'any') if ability.on_phase`

both that ability and the ability I mentioned in the description of the PR work.

Reply:
My initial comment was based on a misunderstanding. Abilities never use `'any'` as an on_phase value, only actual phases or `'never'`.

Reply:
@tobymao I've checked this with several different companies and they all worked as expected.

Reply:
This change broke 18Texas. It uses "on_phase: %w[3 4]," in it's ability definition for several companies.

Reply:
I think 18Texas relied on this not working correctly.

Reply:
So why was this changed again?